hnn_core/optimization.py

```diff
@@ -320,14 +316,17 @@ def _run_optimization(maxiter, param_ranges, optrun):
     return result
 
 
-def optimize_evoked(params, target_dpl, initial_dpl, maxiter=50,
+def optimize_evoked(net_model, params, target_dpl, initial_dpl, maxiter=50,
                     timing_range_multiplier=3.0, sigma_range_multiplier=50.0,
                     synweight_range_multiplier=500.0, decay_multiplier=1.6,
                     scale_factor=1., smooth_window_len=None):
     """Optimize drives to generate evoked response.
 
     Parameters
     ----------
+    net_model : function
+        References the function that instantiates the version of network
+        model used to simulate evoked responses with.
```

Comment:
pass a string ... callable is an advanced Python concept for many people

Or allow passing instance of `Network` but that's more work. Then we'd have to extract the drive parameters from there. It's where we should go eventually

Reply:
Since the available template models will be constantly updated moving forward, I was trying to avoid hardcoding in the possible options.

Reply:
humm ... after a bit of thought, maybe we should do the proper fix while we're at it. That is:

1) instead of `params`, accept instance of `Network`. Make a `copy` like we did for `params`. 
2) iterate over `net.external_drives` and if the drive is of type `evoked`, then 
3) extract the timing and sigma params from `drive['dynamics']` in `_split_by_evinput`
4) use `pick_connection` to get synaptic weights for the drive
5) Then update `_optrun` so that instead of re-instantiating `jones_2009_model` there, you pass `net` and `new_params`. The `new_params` should be used to update `net.external_drves`. Maybe you'll need a private function that does this
6) Finally return the new `net`

sounds doable?

Reply:
okay no worries, let's go with `net_model` then.